src/EnergyPlus/Furnaces.cc

```diff
@@ -8707,6 +8930,8 @@ namespace Furnaces {
 			PartLoadFrac = 0.0;
 		}
 
+		if (true == Furnace(FurnaceNum).bIsIHP) SpeedNum = GetMaxSpeedNumIHP(Furnace(FurnaceNum).CoolingCoilIndex);
```

Comment:
Don't you already know the max speed for each coil? Isn't it SpeedNum = Furnace(FurnaceNum).NumOfSpeed. If not, you should save this during GetInput so you don't have to call this function every time.